### ISSUE TITLE
jdk11 0.17 m2: jtreg failure: java/lang/invoke/InvokeMethodHandleWithBadArgument.java

### ISSUE THREAD

**Issue by andrew-m-leonard** [eclipse/openj9#7396]
Failure link
------------
https://ci.adoptopenjdk.net/job/Test_openjdk11_j9_sanity.openjdk_x86-64_linux/67/console
- test category: openjdk
- test target name, in Grinder equates to TARGET (testCaseName from playlist.xml, jdk_lang, jdk_net, etc): jdk/java/lang/invoke/InvokeMethodHandleWithBadArgument.java
- OS/architecture, in Grinder equates to Jenkinsfile (openjdk_x86-64_linux, openjdk_x86-64_windows, etc):  ALL
- public build SHAs (i.e. java -version output), in Grinder provides JDK_VERSION/JDK_IMPL info and helps identify point at which regression introduced: 
```
13:40:15  openjdk version "11.0.5" 2019-10-15
13:40:15  OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.5+9)
13:40:15  Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.17.0-m2, JRE 11 Linux ppc64le-64-Bit Compressed References 20191007_343 (JIT enabled, AOT enabled)
13:40:15  OpenJ9   - 23b8b11b4
13:40:15  OMR      - 20db4fbc
13:40:15  JCL      - c210c1e1a3 based on jdk-11.0.5+9)
```

Failure output (captured from console output)
---------------------------------------------
```
18:39:37  ACTION: testng -- Failed. Unexpected exit from test [exit code: 255]
18:39:37  REASON: User specified action: run testng/othervm test.java.lang.invoke.InvokeMethodHandleWithBadArgument 
18:39:37  TIME:   3.523 seconds
18:39:37  messages:
18:39:37  command: testng test.java.lang.invoke.InvokeMethodHandleWithBadArgument
18:39:37  reason: User specified action: run testng/othervm test.java.lang.invoke.InvokeMethodHandleWithBadArgument 
18:39:37  Mode: othervm [/othervm specified]
18:39:37  elapsed time (seconds): 3.523
18:39:37  configuration:
18:39:37  STDOUT:
18:39:37  [TestNG] Running:
18:39:37    java/lang/invoke/InvokeMethodHandleWithBadArgument.java
18:39:37  
18:39:37  test test.java.lang.invoke.InvokeMethodHandleWithBadArgument.testAsSpreaderInvokeWithBadElementCount(): success
18:39:37  test test.java.lang.invoke.InvokeMethodHandleWithBadArgument.testAsSpreaderInvokeWithNull(): failure
18:39:37  java.lang.IllegalArgumentException
18:39:37  	at test.java.lang.invoke.InvokeMethodHandleWithBadArgument.testAsSpreaderInvokeWithNull(InvokeMethodHandleWithBadArgument.java:64)
18:39:37  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
18:39:37  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
18:39:37  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
18:39:37  	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
18:39:37  	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
18:39:37  	at org.testng.internal.Invoker.invokeMethod(Invoker.java:639)
18:39:37  	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:821)
18:39:37  	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1131)
18:39:37  	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
18:39:37  	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
18:39:37  	at org.testng.TestRunner.privateRun(TestRunner.java:773)
18:39:37  	at org.testng.TestRunner.run(TestRunner.java:623)
18:39:37  	at org.testng.SuiteRunner.runTest(SuiteRunner.java:357)
18:39:37  	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:352)
18:39:37  	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:310)
18:39:37  	at org.testng.SuiteRunner.run(SuiteRunner.java:259)
18:39:37  	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
18:39:37  	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
18:39:37  	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1185)
18:39:37  	at org.testng.TestNG.runSuitesLocally(TestNG.java:1110)
18:39:37  	at org.testng.TestNG.run(TestNG.java:1018)
18:39:37  	at com.sun.javatest.regtest.agent.TestNGRunner.main(TestNGRunner.java:94)
18:39:37  	at com.sun.javatest.regtest.agent.TestNGRunner.main(TestNGRunner.java:54)
18:39:37  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
18:39:37  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
18:39:37  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
18:39:37  	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
18:39:37  	at com.sun.javatest.regtest.agent.MainWrapper$MainThread.run(MainWrapper.java:127)
18:39:37  	at java.base/java.lang.Thread.run(Thread.java:831)
18:39:37  test test.java.lang.invoke.InvokeMethodHandleWithBadArgument.testAsSpreaderInvokeWithZeroLength(): success
18:39:37  test test.java.lang.invoke.InvokeMethodHandleWithBadArgument.testAsSpreaderPosInvokeWithBadElementCount(): success
18:39:37  test test.java.lang.invoke.InvokeMethodHandleWithBadArgument.testAsSpreaderPosInvokeWithNull(): failure
18:39:37  java.lang.IllegalArgumentException
18:39:37  	at test.java.lang.invoke.InvokeMethodHandleWithBadArgument.testAsSpreaderPosInvokeWithNull(InvokeMethodHandleWithBadArgument.java:57)
18:39:37  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
18:39:37  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
18:39:37  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
18:39:37  	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
18:39:37  	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
18:39:37  	at org.testng.internal.Invoker.invokeMethod(Invoker.java:639)
18:39:37  	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:821)
18:39:37  	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1131)
18:39:37  	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
18:39:37  	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
18:39:37  	at org.testng.TestRunner.privateRun(TestRunner.java:773)
18:39:37  	at org.testng.TestRunner.run(TestRunner.java:623)
18:39:37  	at org.testng.SuiteRunner.runTest(SuiteRunner.java:357)
18:39:37  	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:352)
18:39:37  	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:310)
18:39:37  	at org.testng.SuiteRunner.run(SuiteRunner.java:259)
18:39:37  	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
18:39:37  	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
18:39:37  	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1185)
18:39:37  	at org.testng.TestNG.runSuitesLocally(TestNG.java:1110)
18:39:37  	at org.testng.TestNG.run(TestNG.java:1018)
18:39:37  	at com.sun.javatest.regtest.agent.TestNGRunner.main(TestNGRunner.java:94)
18:39:37  	at com.sun.javatest.regtest.agent.TestNGRunner.main(TestNGRunner.java:54)
18:39:37  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
18:39:37  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
18:39:37  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
18:39:37  	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
18:39:37  	at com.sun.javatest.regtest.agent.MainWrapper$MainThread.run(MainWrapper.java:127)
18:39:37  	at java.base/java.lang.Thread.run(Thread.java:831)
18:39:37  test test.java.lang.invoke.InvokeMethodHandleWithBadArgument.testAsSpreaderPosInvokeWithZeroLength(): success
18:39:37  test test.java.lang.invoke.InvokeMethodHandleWithBadArgument.testExactInvokerWithNull(): success
18:39:37  STDERR:
18:39:37  Unhandled exception
18:39:37  Type=Segmentation error vmState=0x00000000
18:39:37  J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000001
18:39:37  Handler1=00007FF6F11A8A80 Handler2=00007FF6F0A8DE80 InaccessibleAddress=000000000000000C
18:39:37  RDI=0000000001E31300 RSI=0000000000000001 RAX=00007FF6EC013B20 RBX=00007FF6C99DD490
18:39:37  RCX=0000000000000008 RDX=0000000000000000 R8=000000000000003C R9=0000000000000004
18:39:37  R10=0000000000000000 R11=00007FF6F27BFF90 R12=00000000E0141268 R13=0000000001E38310
18:39:37  R14=00000000E0148250 R15=0000000000000002
18:39:37  RIP=00007FF6F117BFD3 GS=0000 FS=0000 RSP=00007FF6C99DD0E0
18:39:37  EFlags=0000000000010246 CS=0033 RBP=0000000000000000 ERR=0000000000000004
18:39:37  TRAPNO=000000000000000E OLDMASK=0000000000000000 CR2=000000000000000C
18:39:37  xmm0 0000000001e31300 (f: 31658752.000000, d: 1.564150e-316)
18:39:37  xmm1 616c2f6176616a4c (f: 1986095744.000000, d: 1.981285e+161)
18:39:37  xmm2 00007ff6c99dd580 (f: 3382564352.000000, d: 6.951401e-310)
18:39:37  xmm3 00007ff6c99dd580 (f: 3382564352.000000, d: 6.951401e-310)
18:39:37  xmm4 00007ff6c99dd580 (f: 3382564352.000000, d: 6.951401e-310)
18:39:37  xmm5 00000000fff475d8 (f: 4294211072.000000, d: 2.121622e-314)
18:39:37  xmm6 00000000fff475c8 (f: 4294211072.000000, d: 2.121622e-314)
18:39:37  xmm7 0000000000000002 (f: 2.000000, d: 9.881313e-324)
18:39:37  xmm8 0000000000000000 (f: 0.000000, d: 0.000000e+00)
18:39:37  xmm9 3fb9773cfd69026f (f: 4251517440.000000, d: 9.947568e-02)
18:39:37  xmm10 3d7546380e210007 (f: 237043712.000000, d: 1.209304e-12)
18:39:37  xmm11 402eb28a9892df05 (f: 2559762176.000000, d: 1.534871e+01)
18:39:37  xmm12 bcb1000000000000 (f: 0.000000, d: -2.359224e-16)
18:39:37  xmm13 bfb9773cfd690280 (f: 4251517440.000000, d: -9.947568e-02)
18:39:37  xmm14 402e7f9c1e980d00 (f: 513281280.000000, d: 1.524924e+01)
18:39:37  xmm15 bc423bc0b7533ac1 (f: 3075685120.000000, d: -1.976870e-18)
18:39:37  Module=/home/jenkins/workspace/Test_openjdk11_j9_sanity.openjdk_x86-64_linux/openjdkbinary/j2sdk-image/lib/compressedrefs/libj9vm29.so
18:39:37  Module_base_address=00007FF6F1109000
18:39:37  Target=2_90_20191007_350 (Linux 4.4.0-142-generic)
18:39:37  CPU=amd64 (4 logical CPUs) (0x1f2f1b000 RAM)
18:39:37  ----------- Stack Backtrace -----------
18:39:37  (0x00007FF6F117BFD3 [libj9vm29.so+0x72fd3])
18:39:37  (0x00007FF6F118386B [libj9vm29.so+0x7a86b])
18:39:37  (0x00007FF6F112327E [libj9vm29.so+0x1a27e])
18:39:37  (0x00007FF6F1121E90 [libj9vm29.so+0x18e90])
18:39:37  (0x00007FF6F11E40B2 [libj9vm29.so+0xdb0b2])
18:39:37  ---------------------------------------
18:39:37  JVMDUMP039I Processing dump event "gpf", detail "" at 2019/10/07 10:39:33 - please wait.
18:39:37  JVMDUMP032I JVM requested System dump using '/home/jenkins/workspace/Test_openjdk11_j9_sanity.openjdk_x86-64_linux/openjdk-tests/TestConfig/test_output_15704696022113/jdk_lang_0/work/scratch/1/core.20191007.103933.14128.0001.dmp' in response to an event
18:39:37  JVMDUMP010I System dump written to /home/jenkins/workspace/Test_openjdk11_j9_sanity.openjdk_x86-64_linux/openjdk-tests/TestConfig/test_output_15704696022113/jdk_lang_0/work/scratch/1/core.20191007.103933.14128.0001.dmp
18:39:37  JVMDUMP032I JVM requested Java dump using '/home/jenkins/workspace/Test_openjdk11_j9_sanity.openjdk_x86-64_linux/openjdk-tests/TestConfig/test_output_15704696022113/jdk_lang_0/work/scratch/1/javacore.20191007.103933.14128.0002.txt' in response to an event
18:39:37  JVMDUMP010I Java dump written to /home/jenkins/workspace/Test_openjdk11_j9_sanity.openjdk_x86-64_linux/openjdk-tests/TestConfig/test_output_15704696022113/jdk_lang_0/work/scratch/1/javacore.20191007.103933.14128.0002.txt
18:39:37  JVMDUMP032I JVM requested Snap dump using '/home/jenkins/workspace/Test_openjdk11_j9_sanity.openjdk_x86-64_linux/openjdk-tests/TestConfig/test_output_15704696022113/jdk_lang_0/work/scratch/1/Snap.20191007.103933.14128.0003.trc' in response to an event
18:39:37  JVMDUMP010I Snap dump written to /home/jenkins/workspace/Test_openjdk11_j9_sanity.openjdk_x86-64_linux/openjdk-tests/TestConfig/test_output_15704696022113/jdk_lang_0/work/scratch/1/Snap.20191007.103933.14128.0003.trc
18:39:37  JVMDUMP007I JVM Requesting JIT dump using '/home/jenkins/workspace/Test_openjdk11_j9_sanity.openjdk_x86-64_linux/openjdk-tests/TestConfig/test_output_15704696022113/jdk_lang_0/work/scratch/1/jitdump.20191007.103933.14128.0004.dmp'
18:39:37  JVMDUMP010I JIT dump written to /home/jenkins/workspace/Test_openjdk11_j9_sanity.openjdk_x86-64_linux/openjdk-tests/TestConfig/test_output_15704696022113/jdk_lang_0/work/scratch/1/jitdump.20191007.103933.14128.0004.dmp
18:39:37  JVMDUMP013I Processed dump event "gpf", detail "".
```

**Comment by DanHeidinga** [comment]
Tagging this as high priority as it produces a segfault

**Comment by pshipton** [comment]
@DanHeidinga this is a failure in the 0.17 jdk11 milestone 2 build. Did you intend to tag against the 0.18 release, or is the intention to fix this for the 0.17 release?

**Comment by pshipton** [comment]
I expect it's a regression since the milestone 1 build, as we didn't see this crash then.

**Comment by pshipton** [comment]
I think this is a dup of https://github.com/eclipse/openj9/issues/6943, for which the fix wasn't delivered to the 0.17 release.

**Comment by DanHeidinga** [comment]
Retargeted to `0.17.0` as the change in #7062 is safe to include in this release. 

FYI @fengxue-IS this is the issue to use in the `fixed:` tag in the 0.17 backport

**Comment by pshipton** [comment]
The fix was merged to the 0.17 release.
